cogboard-app/src/main/kotlin/com/cognifide/cogboard/logStorage/LogStorage.kt

```diff
@@ -0,0 +1,293 @@
+    get() = database?.getCollection(config.id)
+
+    // Storage configuration
+
+    /** Returns a logs collection configuration associated with this widget (if present). */
+    private val collectionConfiguration: LogCollectionConfiguration?
+    get() = configCollection
+            ?.find(eq(Log.ID, config.id))
+            ?.first()
+            ?.let { LogCollectionConfiguration.from(it) }
+
+    /** Saves or deletes a logs collection [configuration] associated with this widget. */
+    private fun saveConfiguration(configuration: LogCollectionConfiguration?) {
+        if (configuration != null) {
+            val options = ReplaceOptions().upsert(true)
+            configCollection
+                    ?.replaceOne(
+                            eq(LogCollectionConfiguration.ID, config.id),
+                            configuration.toDocument(),
+                            options
+                    )
+        } else {
+            configCollection?.deleteOne((eq(LogCollectionConfiguration.ID, config.id)))
+        }
+    }
+
+    // MongoDB - logs
+
+    /** Deletes all logs from the collection. */
+    private fun deleteAllLogs() {
+        logsCollection?.drop()
+    }
+
+    /** Deletes the [n] first logs (ordered by their sequence number). */
+    private fun deleteFirstLogs(n: Long) {
+        val collection = logsCollection ?: return
+        val ids = collection
+                .find()
+                .sort(ascending(Log.SEQ))
+                .limit(n.toInt())
+                .mapNotNull { it?.getObjectId(Log.ID) }
+        if (ids.isNotEmpty()) {
+            try {
+                val result = collection.deleteMany(`in`(Log.ID, ids))
+                LOGGER.debug("Deleted ${result.deletedCount} first logs")
+            } catch (exception: MongoException) {
+                LOGGER.error("Cannot delete first logs: $exception")
+            }
+        }
+    }
+
+    /** Deletes old logs (based on the number of days before expiration). */
+    private fun deleteOldLogs() {
+        val collection = logsCollection ?: return
+        val now = Instant.now().epochSecond
+        val beforeTimestamp = now - (config.expirationDays * DAY_TO_TIMESTAMP)
+        try {
+            val result = collection.deleteMany(lt(Log.INSERTED_ON, beforeTimestamp))
+            LOGGER.debug("Deleted ${result.deletedCount} old logs")
+        } catch (exception: MongoException) {
+            LOGGER.error("Cannot delete old logs: $exception")
+        }
+    }
+
+    /** Deletes oldest logs when logs take up too much space. */
+    private fun deleteSpaceConsumingLogs() {
+        val database = database ?: return
+        val collection = logsCollection ?: return
+
+        val size = database
+                .runCommand(Document(STATS_COMMAND, config.id))
+                .getInteger(STATS_SIZE) ?: 0
+        val maxSize = config.fileSizeMB * MB_TO_BYTES
+        if (size > 0 && size > maxSize) {
```

Comment:
why do we need to check if it's more than `0`?

Reply:
We divide by `size` later.